bot/exts/evergreen/connect_four.py

```diff
@@ -0,0 +1,440 @@
+        aliases=["4inarow", "connect4", "connectfour", "c4"],
+        case_insensitive=True
+    )
+    async def connect_four(
+            self,
+            ctx: commands.Context,
+            board_size: int = 7,
+            emoji1: EMOJI_CHECK = "\U0001f535",
+            emoji2: EMOJI_CHECK = "\U0001f534"
+    ) -> None:
+        """
+        Play the classic game of Connect Four with someone!
+
+        Sets up a message waiting for someone else to react and play along.
+        The game will start once someone has reacted.
+        All inputs will be through reactions.
+        """
+        if isinstance(emoji1, str) and len(emoji1) > 1:
+            raise commands.EmojiNotFound(emoji1)
+        if isinstance(emoji2, str) and len(emoji2) > 1:
+            raise commands.EmojiNotFound(emoji2)
+
+        check_author_result = await self.check_author(ctx, board_size)
+        if not check_author_result:
+            return
+
+        announcement = await ctx.send(
+            "**Connect Four**: A new game is about to start!\n"
+            f"Press {HAND_RAISED_EMOJI} to play against {ctx.author.mention}!\n"
+            f"(Cancel the game with {CROSS_EMOJI}.)"
+        )
+        self.waiting.append(ctx.author)
+        await announcement.add_reaction(HAND_RAISED_EMOJI)
+        await announcement.add_reaction(CROSS_EMOJI)
+
+        try:
+            reaction, user = await self.bot.wait_for(
+                "reaction_add",
+                check=partial(self.get_player, ctx, announcement),
+                timeout=60.0
+            )
+        except asyncio.TimeoutError:
+            self.waiting.remove(ctx.author)
+            await announcement.delete()
+            await ctx.send(
+                f"{ctx.author.mention} Seems like there's no one here to play. "
+                f"Use `{ctx.prefix}{ctx.invoked_with} ai` to play against a computer."
+            )
+            return
+
+        if str(reaction.emoji) == CROSS_EMOJI:
+            self.waiting.remove(ctx.author)
+            await announcement.delete()
+            await ctx.send(f"{ctx.author.mention} Game cancelled.")
+            return
+
+        await announcement.delete()
+        self.waiting.remove(ctx.author)
+        if self.already_playing(ctx.author):
+            return
+
+        await self._play_game(ctx, user, board_size, str(emoji1), str(emoji2))
+
+    @connect_four.command(aliases=["bot", "computer", "cpu"])
+    async def ai(
+            self,
+            ctx: commands.Context,
+            board_size: int = 7,
+            emoji1: EMOJI_CHECK = "\U0001f535"
+    ) -> None:
+        """Play Connect Four against a computer player."""
+        if isinstance(emoji1, str) and len(emoji1) > 1:
+            raise commands.EmojiNotFound(emoji1)
+        check_author_result = await self.check_author(ctx, board_size)
+        if not check_author_result:
+            return
+        await self._play_game(ctx, None, board_size, str(emoji1), ":red_circle:")
```

Comment:
I don't see a reason why we can't allow the player to set the bot's token as well.